awslabs/aws-lc
# Basic Framework for FIPS Service Indicator

### Issues:
Addresses `CryptoAlg-835` 

### Description of changes: 
Basic framework and tests of Service Indicator for FIPS 140-3 validation.

### Call-outs:
Indicators checks have only been added for the `aead` method for AES-GCM and one-shot function for AES-CBC in this PR so far. 
The main focus of this PR is to look at the correctness of the framework of the service indicator, but feel free to give advice on where the indicator checks that exist right now could be better placed.

### Testing:
Added new `service_indicator.cc` dimension:

- Basic function test
- Check for approval of AES-CBC one-shot function
- Check for approval of AES-GCM `aead` API usage


By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
